### PR TITLE
refactor: change API properties names to snake_case

### PR DESCRIPTION
**Pull request by imincik**
Closes: #685 

## Reference

[Code review checklist](/linz/geospatial-data-lake/blob/master/CODING.md#Code-review-checklist)

### CODE REVIEW

File: USAGE.md [hunk 1]
```diff
@@ -187,8 +187,8 @@ aws lambda invoke \
   ```console
   $ aws lambda invoke \
      --function-name "${ENV}-import-status" \
-     --payload '{"httpMethod": "GET", "body": {"execution_arn": "arn:aws:batch:ap-southeast-2:xxxx:job/example-arn"}}' \
+     --payload '{"http_method": "GET", "body": {"execution_arn": "arn:aws:batch:ap-southeast-2:xxxx:job/example-arn"}}' \
      /dev/stdout
 
-  {"statusCode": 200, "body": {"validation":{ "status": "SUCCEEDED"}, "metadata upload":{"status": "Pending", "errors":[]}, "asset upload":{"status": "Pending", "errors":[]}}}
+  {"status_code": 200, "body": {"validation":{ "status": "SUCCEEDED"}, "metadata upload":{"status": "Pending", "errors":[]}, "asset upload":{"status": "Pending", "errors":[]}}}
```
Comment (l0b0) [inline USAGE.md:193]: `metadata upload` should probably be `metadata_upload`, and `asset upload` should be `asset_upload`.

File: tests/test_dataset_versions_endpoint.py [hunk 2]
```diff
@@ -52,11 +56,13 @@ def should_return_required_property_error_when_missing_mandatory_id_property() -
 def should_return_error_if_dataset_id_does_not_exist_in_db() -> None:
     body = {"id": any_dataset_id(), "metadata-url": any_s3_url()}
```
Comment (l0b0) [inline tests/test_dataset_versions_endpoint.py:57]: `metadata-url` should be `metadata_url`.

**Review by l0b0 — CHANGES_REQUESTED** [review]

File: USAGE.md (hunk 1 above)
Comment (imincik) [reply USAGE.md:193]: Thank you @l0b0 . I'll fix that.

File: tests/test_dataset_versions_endpoint.py (hunk 2 above)
Comment (imincik) [reply tests/test_dataset_versions_endpoint.py]: Should be fixed in 79753324dbc409a9d501a77203fc820c27be2c46

File: USAGE.md (hunk 1 above)
Comment (imincik) [reply USAGE.md]: Should be fixed in 68ac4de5d1bc28b22b92d10ba9169953060f3e7b

**Comment by imincik** [comment]
> There are more non-standard keys in `backend/import_dataset_keys.py` and `backend/error_response_keys.py`.

Should be fixed in f62db44dc41212eefccd8f103903b73fb161881e

**Review by l0b0 — APPROVED** [review]

**Pull request merged**